redcap_bridge/utils.py

```diff
@@ -81,3 +81,27 @@ def remove_columns(csv_file, compressed_file=None):
 def exportCSVtoXLS(csv_file, compressed_file=None):
     read_file = pd.read_csv(csv_file, na_filter=False, dtype='str')
     read_file.to_excel(r'Path', index=None, header=True)
+
+def conversion_to_odml_table_descriptor(full_elabbook_csv, session_number):
+    """
+        Create odml descriptor file base on the full elabbook csv file
+        Args:
+            full_elabbook_csv: path to the full csv file
+            session_number: number of the session you want to create descriptor csv file
+        Returns:
+            descriptor_elabbook_csv: csv descriptor file of the specific session given
+        """
+
+    df = pd.read_csv(full_elabbook_csv)
+
+    df_ses = df.loc[df['ses_number'] == session_number]
+
+    cols_to_melt = df_ses.columns[df_ses.columns.get_loc('ethical_protocol_id'):]
+
+    print(cols_to_melt)
+
+    df_melted = df_ses.melt(id_vars='record_id', value_vars=cols_to_melt)
+
+    df_melted = df_melted.rename(columns={'variable': 'Property name'})
+
+    print(df_melted.to_string())
```

Comment:
There is a set of mandatory columns for odmltables csv files. Can you check that all of them are present after the conversion and have values that are accepted by odmltables? E.g. the data type colum can only contain specific values that are understood by odmltables...